store/precomputed_key/wvm/rpc/wvm_rpc_client.go

```diff
@@ -0,0 +1,241 @@
+	GetAccount(ctx context.Context) (common.Address, error)
+	SignTransaction(ctx context.Context, signData *wvmtypes.SignData) (string, error)
+}
+
+// WVM RPC client
+type RPCClient struct {
+	log     log.Logger
+	client  *ethclient.Client
+	chainID int64
+	signer  Signer
+}
+
+func NewWvmRPCClient(log log.Logger, cfg *wvmtypes.Config, signer Signer) (*RPCClient, error) {
+	client, err := ethclient.Dial(cfg.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to the WVM client: %w", err)
+	}
+
+	ethRPCClient := &RPCClient{
+		log:     log,
+		client:  client,
+		chainID: cfg.ChainID,
+		signer:  signer,
+	}
+
+	return ethRPCClient, nil
+}
+
+func (rpc *RPCClient) SendTransaction(ctx context.Context, to string, data []byte) (string, error) {
+	gas, err := rpc.estimateGas(ctx, to, data)
+	if err != nil {
+		return "", fmt.Errorf("failed to store data in wvm: failed estimate gas: %w", err)
+	}
+
+	wvmRawTx, err := rpc.createRawTransaction(ctx, to, string(data), gas)
+	if err != nil {
+		return "", fmt.Errorf("failed to store data in wvm: failed create transaction: %w", err)
+	}
+
+	wvmTxHash, err := rpc.sendRawTransaction(ctx, wvmRawTx)
+	if err != nil {
+		return "", fmt.Errorf("failed to store data in wvm: failed to send transaction: %w", err)
+	}
+
+	return wvmTxHash, nil
+}
+
+// estimateGas tries estimates the suggested amount of gas that required to execute a given transaction.
+func (rpc *RPCClient) estimateGas(ctx context.Context, to string, data []byte) (uint64, error) {
+	var (
+		toAddr    = common.HexToAddress(to)
+		bytesData []byte
+		err       error
+	)
+
+	fromAddress, err := rpc.signer.GetAccount(ctx)
+	if err != nil {
+		return 0, fmt.Errorf("failed to estimate gas, no signer")
+	}
+
+	var encoded string
+	if string(data) != "" {
+		if ok := strings.HasPrefix(string(data), "0x"); !ok {
+			encoded = hexutil.Encode(data)
+		}
+
+		bytesData, err = hexutil.Decode(encoded)
+		if err != nil {
+			return 0, err
+		}
+	}
+
+	msg := ethereum.CallMsg{
+		From: fromAddress,
+		To:   &toAddr,
+		Gas:  0x00,
+		Data: bytesData,
+	}
+
+	gas, err := rpc.client.EstimateGas(ctx, msg)
+	if err != nil {
+		return 0, err
+	}
+
+	rpc.log.Info("wvm: estimated tx gas price", "price", gas)
+
+	return gas, nil
+}
+
+// createRawTransaction creates a raw EIP-1559 transaction and returns it as a hex string.
+func (rpc *RPCClient) createRawTransaction(ctx context.Context, to string, data string, gasLimit uint64) (string, error) {
+	baseFee, err := rpc.client.SuggestGasPrice(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	fromAddress, err := rpc.signer.GetAccount(ctx)
+	if err != nil {
+		return "", fmt.Errorf("failed to get an account from signer: %w", err)
+	}
+	nonce, err := rpc.client.PendingNonceAt(ctx, fromAddress)
+	if err != nil {
+		return "", err
+	}
+
+	signData := wvmtypes.SignData{To: to, Data: data, GasLimit: gasLimit, GasFeeCap: baseFee, Nonce: nonce}
+	return rpc.signer.SignTransaction(ctx, &signData)
```

Comment:
are there async collisions that can occur if this function is called twice in parallel? e.g, incorrect nonce mgmt

Reply:
checked, 
nonce from ethclient.PendingNonceAt call should be pretty reliable cause considers 'pending' transactions too
and if there will be any error during "send" the retry mechanism can repeat and resolve it
iiuc daproxy shouldn't have load of hundreds/thousands blobs which will transform in hundreds/thousands txs for the particular address